solutuins-1/src/main/java/solutions_1/solutuins_1/KattapaReadyForBattle.java

```diff
@@ -0,0 +1,52 @@
+package solutions_1.solutuins_1;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * The Class KattapaReadyForBattle.
+ */
+
+public class KattapaReadyForBattle {
+
+	/** The logger. */
+	private Logger logger;
+
+	/**
+	 * Instantiates a new KattapaReadyForBattle
+	 */
+	public KattapaReadyForBattle() {
+		logger = Logger.getLogger(KattapaReadyForBattle.class.getSimpleName());
+	}
+
+	/**
+	 * Checks if the even weapon solders are greater than odd weapon soldiers.
+	 *
+	 * @param int size of the army, Integer[] number of weapons each soldier is carrying
+	 * @return String ready or not ready for battle based on Kattapa's luck logic
+	 */
+
+	public String isReadyOrNot(int size, Integer[] soldierWeapons) {

```

Comment:
why an array of Integer not int?

Reply:
No particular reason here, can use int as well. In PrimeString problem I had used Maps which didn't accept the primitive datatype int so I was using Integer there. In that flow I had used Integer here.

Reply:
Can you please explain difference between both with advantages and disadvantages. Cant we use int in map directly instead of Integer?

Reply:
Integer is a wrapper class right, so it has many additional methods. Primitive datatypes are just basic. Advantage of a primitive datatype is that performance wise it is faster compared to Wrapper classes, and wrapper classes takes up more space compared to primitive datatypes.
We can't use int directly in the map because it accepts only classes derived from Object, int, float etc., primitives are not derived from Object.